src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -281,9 +280,19 @@ internal async Task ProcessMessage(Message receivedMessage, CancellationToken me
 
             try
             {
+                string messageId;
+                if (receivedMessage.MessageAttributes.TryGetValue(Headers.MessageId, out var messageIdAttribute))
+                {
+                    messageId = messageIdAttribute.StringValue;
+                }
+                else
+                {
+                    messageId = nativeMessageId;
+                }
+
```

Comment:
What about moving this into a separate static method so that it can be called by tests too?